src/components/AnnouncementsModal.js

```diff
@@ -0,0 +1,305 @@
+import { Icon, Icons } from './publiq-ui/Icon';
+import { Image } from './publiq-ui/Image';
+import { Inline } from './publiq-ui/Inline';
+import { Link } from './publiq-ui/Link';
+import { List } from './publiq-ui/List';
+import { ListItem } from './publiq-ui/ListItem';
+import { ModalContent } from './publiq-ui/ModalContent';
+import { Stack } from './publiq-ui/Stack';
+import { getValueFromTheme } from './publiq-ui/theme';
+import { Title } from './publiq-ui/Title';
+
+const getValueForAnnouncementItem = getValueFromTheme('announcementItem');
+const getValueForAnnouncementList = getValueFromTheme('announcementList');
+const getValueForAnnouncementContent = getValueFromTheme('announcementContent');
+
+const AnnouncementItem = ({
+  id,
+  title,
+  selected,
+  seen,
+  setSelectedAnnouncementId,
+}) => {
+  return (
+    <ListItem
+      padding={4}
+      spacing={3}
+      css={`
+        cursor: pointer;
+        border-bottom: ${getValueForAnnouncementItem('borderColor')} 1px solid;
+        background-color: ${selected
+          ? getValueForAnnouncementItem('selected.backgroundColor')
+          : 'inherit'};
+
+        :hover {
+          background-color: ${selected
+            ? getValueForAnnouncementItem('selected.hoverBackgroundColor')
+            : getValueForAnnouncementItem('hoverBackgroundColor')};
+        }
+      `}
+      onClick={() => {
+        setSelectedAnnouncementId(id);
+      }}
+    >
+      {seen ? <Icon name={Icons.EYE_SLASH} /> : <Icon name={Icons.EYE} />}
+      <Box
+        as="span"
+        css={`
+          white-space: nowrap;
+          overflow: hidden;
+          text-overflow: ellipsis;
+        `}
+      >
+        {title}
+      </Box>
+    </ListItem>
+  );
+};
+
+AnnouncementItem.propTypes = {
+  id: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  selected: PropTypes.bool,
+  seen: PropTypes.bool,
+  setSelectedAnnouncementId: PropTypes.func,
+};
+
+AnnouncementItem.defaultProps = {
+  selected: false,
+  seen: false,
+};
+
+const AnnouncementsList = ({
+  announcements,
+  selectedAnnouncementId,
+  seenAnnouncements,
+  setSelectedAnnouncementId,
+}) => (
+  <List
+    css={`
+      width: 30%;
+      border-right: ${getValueForAnnouncementList('borderColor')} 1px solid;
+      overflow-y: auto;
+    `}
+  >
+    {announcements.map((announcement) => {
+      return (
+        <AnnouncementItem
+          key={announcement.uid}
+          id={announcement.uid}
+          title={announcement.title}
+          selected={announcement.uid === selectedAnnouncementId}
+          seen={seenAnnouncements.includes(announcement.uid)}
+          setSelectedAnnouncementId={setSelectedAnnouncementId}
+        />
+      );
+    })}
+  </List>
+);
+
+AnnouncementsList.propTypes = {
+  announcements: PropTypes.array,
+  selectedAnnouncementId: PropTypes.string,
+  seenAnnouncements: PropTypes.array,
+  setSelectedAnnouncementId: PropTypes.func,
+};
```

Comment:
Is it usefull to wrap up this in another component? would simplify a lot of the 'prop drilling'